tests/plugins/test_deepspeed_plugin.py

```diff
@@ -0,0 +1,318 @@
+import json
+import os
+
+import pytest
+import torch
+from torch import Tensor
+from torch.optim import Optimizer
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.plugins import DeepSpeedPlugin, DeepSpeedPrecisionPlugin
+from pytorch_lightning.utilities import _APEX_AVAILABLE, _DEEPSPEED_AVAILABLE, _NATIVE_AMP_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+
+
+@pytest.fixture
+def deepspeed_config():
+    return {
+        "optimizer": {
+            "type": "SGD",
+            "params": {
+                "lr": 3e-5,
+            },
+        },
+        'scheduler': {
+            "type": "WarmupLR",
+            "params": {
+                "last_batch_iteration": -1,
+                "warmup_min_lr": 0,
+                "warmup_max_lr": 3e-5,
+                "warmup_num_steps": 100,
+            }
+        }
+    }
+
+
+@pytest.mark.skipif(not _DEEPSPEED_AVAILABLE, reason="DeepSpeed not available.")
+def test_deepspeed_plugin_string(tmpdir):
+    """
+        Test to ensure that the plugin can be passed via string, and parallel devices is correctly set.
+    """
+
+    class CB(Callback):
+
+        def on_fit_start(self, trainer, pl_module):
+            assert isinstance(trainer.accelerator_backend.training_type_plugin, DeepSpeedPlugin)
+            assert trainer.accelerator_backend.training_type_plugin.parallel_devices == [torch.device('cpu')]
+            raise SystemExit()
+
+    model = BoringModel()
+    trainer = Trainer(
+        fast_dev_run=True,
+        plugins='deepspeed',
+        callbacks=[CB()],
+    )
+
+    with pytest.raises(SystemExit):
+        trainer.fit(model)
```

Comment:
is it really necessary to enter fit and systemexit in the callback?
I believe for these asserts it is sufficient to simply instantiate trainer, since the accelerator is fully configured in the init of the trainer.

Reply:
Nice catch, updated